native/src/main/java/io/ballerina/stdlib/http/api/HttpResource.java

```diff
@@ -92,6 +101,38 @@ protected HttpResource() {
 
     }
 
+    public String getResourceLinkName() {
+        return resourceLinkName;
+    }
+
+    public void setResourceLinkName(String name) {
+        this.resourceLinkName = name;
+    }
+
+    public List<LinkedResourceInfo> getLinkedResources() {
+        return linkedResources;
+    }
+
+    public void addLinkedResource(LinkedResourceInfo linkedResourceInfo) {
+        this.linkedResources.add(linkedResourceInfo);
+    }
+
+    public void addLink(BString relation, BMap link) {
+        this.links.put(relation, link);
+    }
+
+    public BMap<BString, Object> getLinks() {
+        return this.links;
+    }
+
+    public boolean hasLinkedRelation(BString relation) {
+        if (this.linkedRelations.contains(relation)) {
+            return true;
+        }
+        this.linkedRelations.add(relation);
+        return false;
+    }
```

Comment:
The method name implies it checks whether particular relation is already exists. But internally we are adding the relation if it is not there. Could you please refactor this properly bcos this breaks SRP.

Reply:
Fixed in [a8c835e](https://github.com/ballerina-platform/module-ballerina-http/pull/1068/commits/a8c835e645db41d31e5f041b2fc786856153dbb3)